dbt_meshify/utilities/versioner.py

```diff
@@ -134,19 +136,21 @@ def bump_version(
         path = self.project.resolve_patch_path(model)
         model_path = self.project.path / model.original_file_path
 
-        try:
-            model_yml = self.load_model_yml(path, model.name)
+        model_yml = self.load_model_yml(path, model.name)
+        if model_override:
+            model_yml = safe_update(model_yml, model_override)
 
-            # If a model override has been provided safely update the `model_yml` with the override.
-            if model_override:
-                model_yml = safe_update(model_yml, model_override)
-        except FileNotFoundError:
-            model_yml = {}
-
-        latest_version = model_yml.get("latest_version", 0)
         model_versions: NamedList = self.get_model_versions(model_yml)
         greatest_version = self.get_latest_model_version(model_versions)
 
+        if len(model_versions) == 0:
+            raise ModelVersionerException(
+                f"The model {model.name} does not have any versions defined. Please use add-version first."
+            )
+
+        # Within dbt-core, if unset `latest_version` defaults to the greatest version identifier.
+        latest_version = model_yml.get("latest_version", greatest_version)
+
         # Bump versions
         new_greatest_version_number = greatest_version + 1
         new_latest_version_number = latest_version if prerelease else latest_version + 1
```

Comment:
I think this line may be the source of the error -- when bumping with prerelease from this yml:

```yml
models:
  - name: shared_model
    latest_version: 1
    description: "this is a test model"
    versions:
      - v: 1
```

we'd expect the `latest_version` to remain `1` but to have a new file at `shared_model_v2`. 

the filepath calc a few lines down is 

```
next_version_file_name = model_path.parent / Path(
            f"{defined_in}.{model.language}"
            if defined_in
            else f"{model.name}_v{new_latest_version_number}.{model.language}"
        )
```

which in this case would be `shared_model_v1.sql` which already exists! I think it's only at this first point that this is an issue, so we might just need to add custom logic for when this is your starting point?

Reply:
the fix might be as simple as using `new_greatest_version_number`  for the filepath calculation?

Reply:
100000% agreed